{
    "url": "https://api.github.com/repos/pallets/flask/issues/3361",
    "repository_url": "https://api.github.com/repos/pallets/flask",
    "labels_url": "https://api.github.com/repos/pallets/flask/issues/3361/labels{/name}",
    "comments_url": "https://api.github.com/repos/pallets/flask/issues/3361/comments",
    "events_url": "https://api.github.com/repos/pallets/flask/issues/3361/events",
    "html_url": "https://github.com/pallets/flask/issues/3361",
    "id": 492183908,
    "node_id": "MDU6SXNzdWU0OTIxODM5MDg=",
    "number": 3361,
    "title": "Custom command for CLI does not raise import exceptions",
    "user": {
        "login": "darkclouder",
        "id": 2065342,
        "node_id": "MDQ6VXNlcjIwNjUzNDI=",
        "avatar_url": "https://avatars3.githubusercontent.com/u/2065342?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/darkclouder",
        "html_url": "https://github.com/darkclouder",
        "followers_url": "https://api.github.com/users/darkclouder/followers",
        "following_url": "https://api.github.com/users/darkclouder/following{/other_user}",
        "gists_url": "https://api.github.com/users/darkclouder/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/darkclouder/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/darkclouder/subscriptions",
        "organizations_url": "https://api.github.com/users/darkclouder/orgs",
        "repos_url": "https://api.github.com/users/darkclouder/repos",
        "events_url": "https://api.github.com/users/darkclouder/events{/privacy}",
        "received_events_url": "https://api.github.com/users/darkclouder/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [],
    "state": "closed",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": null,
    "comments": 3,
    "created_at": "2019-09-11T11:39:54Z",
    "updated_at": "2019-11-06T09:38:17Z",
    "closed_at": "2019-10-12T16:50:08Z",
    "author_association": "NONE",
    "body": "**Problem:**\r\nCreating a custom command for the flask CLI with an import exception is silent, even when running the command, e.g. `start` with `flask start` directly.\r\nThis makes it difficult to debug import errors in changing environments, as you're left with\r\n```\r\nError: No such command \"start\".\r\n```\r\nwithout further explanation.\r\n\r\n**Origin:**\r\nhttps://github.com/pallets/flask/blob/d694eb443a6243e248df974e9cd64defb591bf38/src/flask/cli.py#L541-L546\r\n`NoAppException` is caught silently with a `pass`, even though `get_command` seems to only be used to execute a single command rather than an entire command listing.\r\nHowever, even the command listing of `list_commands` explicitly prints the traceback with `traceback.print_exc()`.\r\nWhy that is not done in `get_command` is not clear to me at this point.",
    "comments_inline": [
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/541342108",
            "html_url": "https://github.com/pallets/flask/issues/3361#issuecomment-541342108",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/3361",
            "id": 541342108,
            "node_id": "MDEyOklzc3VlQ29tbWVudDU0MTM0MjEwOA==",
            "user": {
                "login": "davidism",
                "id": 1242887,
                "node_id": "MDQ6VXNlcjEyNDI4ODc=",
                "avatar_url": "https://avatars1.githubusercontent.com/u/1242887?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/davidism",
                "html_url": "https://github.com/davidism",
                "followers_url": "https://api.github.com/users/davidism/followers",
                "following_url": "https://api.github.com/users/davidism/following{/other_user}",
                "gists_url": "https://api.github.com/users/davidism/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/davidism/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/davidism/subscriptions",
                "organizations_url": "https://api.github.com/users/davidism/orgs",
                "repos_url": "https://api.github.com/users/davidism/repos",
                "events_url": "https://api.github.com/users/davidism/events{/privacy}",
                "received_events_url": "https://api.github.com/users/davidism/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2019-10-12T16:50:05Z",
            "updated_at": "2019-10-12T16:50:05Z",
            "author_association": "MEMBER",
            "body": "Duplicate of #2741"
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/542242228",
            "html_url": "https://github.com/pallets/flask/issues/3361#issuecomment-542242228",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/3361",
            "id": 542242228,
            "node_id": "MDEyOklzc3VlQ29tbWVudDU0MjI0MjIyOA==",
            "user": {
                "login": "darkclouder",
                "id": 2065342,
                "node_id": "MDQ6VXNlcjIwNjUzNDI=",
                "avatar_url": "https://avatars3.githubusercontent.com/u/2065342?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/darkclouder",
                "html_url": "https://github.com/darkclouder",
                "followers_url": "https://api.github.com/users/darkclouder/followers",
                "following_url": "https://api.github.com/users/darkclouder/following{/other_user}",
                "gists_url": "https://api.github.com/users/darkclouder/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/darkclouder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/darkclouder/subscriptions",
                "organizations_url": "https://api.github.com/users/darkclouder/orgs",
                "repos_url": "https://api.github.com/users/darkclouder/repos",
                "events_url": "https://api.github.com/users/darkclouder/events{/privacy}",
                "received_events_url": "https://api.github.com/users/darkclouder/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2019-10-15T14:31:11Z",
            "updated_at": "2019-10-15T14:31:11Z",
            "author_association": "NONE",
            "body": "The linked ticket has nothing to do with this issue. Please reopen.\r\nEnvironment variables are set, this is regarding an import error within the app in the environment variable."
        },
        {
            "url": "https://api.github.com/repos/pallets/flask/issues/comments/550227938",
            "html_url": "https://github.com/pallets/flask/issues/3361#issuecomment-550227938",
            "issue_url": "https://api.github.com/repos/pallets/flask/issues/3361",
            "id": 550227938,
            "node_id": "MDEyOklzc3VlQ29tbWVudDU1MDIyNzkzOA==",
            "user": {
                "login": "darkclouder",
                "id": 2065342,
                "node_id": "MDQ6VXNlcjIwNjUzNDI=",
                "avatar_url": "https://avatars3.githubusercontent.com/u/2065342?v=4",
                "gravatar_id": "",
                "url": "https://api.github.com/users/darkclouder",
                "html_url": "https://github.com/darkclouder",
                "followers_url": "https://api.github.com/users/darkclouder/followers",
                "following_url": "https://api.github.com/users/darkclouder/following{/other_user}",
                "gists_url": "https://api.github.com/users/darkclouder/gists{/gist_id}",
                "starred_url": "https://api.github.com/users/darkclouder/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/darkclouder/subscriptions",
                "organizations_url": "https://api.github.com/users/darkclouder/orgs",
                "repos_url": "https://api.github.com/users/darkclouder/repos",
                "events_url": "https://api.github.com/users/darkclouder/events{/privacy}",
                "received_events_url": "https://api.github.com/users/darkclouder/received_events",
                "type": "User",
                "site_admin": false
            },
            "created_at": "2019-11-06T09:38:17Z",
            "updated_at": "2019-11-06T09:38:17Z",
            "author_association": "NONE",
            "body": "@davidism please reopen."
        }
    ]
}